test/ZNSTreasury.test.ts

```diff
@@ -61,16 +61,21 @@ describe("ZNSTreasury", () => {
     zns = await deployZNS(params);
 
     paymentConfig = {
-      token: await zns.meowToken.getAddress(),
+      token: await zns.zToken.getAddress(),
       beneficiary: user.address,
     };
 
     // give REGISTRAR_ROLE to a wallet address to be calling guarded functions
     await zns.accessController.connect(admin).grantRole(REGISTRAR_ROLE, mockRegistrar.address);
 
     // Give funds to user
-    await zns.meowToken.connect(user).approve(await zns.treasury.getAddress(), ethers.MaxUint256);
-    await zns.meowToken.mint(user.address, ethers.parseEther("50000"));
+    const userBalanceInitial = await zns.zToken.balanceOf(admin.address) / 3n;
+    await zns.zToken.connect(admin).approve(await zns.treasury.getAddress(), userBalanceInitial);
```

Comment:
This code is strange to me. Why is the admin calling to approve the treasury here? We only need the `user` to approve the treasury for when they call to register below. To simply transfer token from `admin` to `user` we don't need any approval. There should be just two calls, one for the user to approve the treasury, and one for the admin to transfer funds to the user

Reply:
agree. there are a lot of admin approvals of registry in this whole file. I believe none of them are needed.